internal/pkg/agent/application/inject_proxy_component_modifier.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"os"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator"
+	"github.com/elastic/elastic-agent/pkg/component"
+)
+
+// InjectProxyComponentModifier injects a proxy_url value into a component's config if one is not set.
+//
+// This will inject a proxy_url into each output unit if there is no existing value and the agent has a proxy set through an environment variable.
+// The URL used is the HTTPS_PROXY env var. If that's not set the HTTP_PROXY env var is used.
+// If there are no env vars set, or the unit's config has `proxy_disable: true`, nothing is injected
+func InjectProxyComponentModifier() coordinator.ComponentsModifier {
+	proxyURL, ok := os.LookupEnv("HTTPS_PROXY")
+	if !ok {
+		proxyURL, _ = os.LookupEnv("HTTP_PROXY")
+	}
+	return func(comps []component.Component, cfg map[string]interface{}) ([]component.Component, error) {
+		if proxyURL == "" {
+			return comps, nil
+		}
+		for i, comp := range comps {
+			if comp.InputSpec != nil {
+				for j, unit := range comp.Units {
+					if unit.Type == client.UnitTypeOutput {
```

Comment:
This is going to apply to all output types not just elasticsearch. Based on the logic here I believe this should be elasticsearch specific? Would we have a logstash one?